www/__tests__/customMetricsHelper.test.ts

```diff
@@ -0,0 +1,51 @@
+import { getConfig } from '../js/config/dynamicConfig';
+import {
+  getCustomFootprint,
+  getCustomMETs,
+  initCustomDatasetHelper,
+} from '../js/metrics/customMetricsHelper';
+import { mockBEMUserCache } from '../__mocks__/cordovaMocks';
+import { mockLogger } from '../__mocks__/globalMocks';
+import fakeLabels from '../__mocks__/fakeLabels.json';
+
+mockBEMUserCache();
+mockLogger();
+
+global.fetch = (url: string) =>
+  new Promise((rs, rj) => {
+    setTimeout(() =>
+      rs({
+        text: () =>
+          new Promise((rs, rj) => {
+            let myJSON = JSON.stringify(fakeLabels);
+            setTimeout(() => rs(myJSON), 100);
+          }),
+      }),
+    );
+  }) as any;
+
+it('gets the custom mets', async () => {
+  initCustomDatasetHelper(getConfig());
+  await new Promise((r) => setTimeout(r, 800));
+  expect(getCustomMETs()).toMatchObject({
+    walk: {},
+    bike: {},
+    bikeshare: {},
+    'e-bike': {},
+    scootershare: {},
+    drove_alone: {},
+  });
+});
+
+it('gets the custom footprint', async () => {
+  initCustomDatasetHelper(getConfig());
+  await new Promise((r) => setTimeout(r, 800));
+  expect(getCustomFootprint()).toMatchObject({
+    walk: {},
+    bike: {},
+    bikeshare: {},
+    'e-bike': {},
+    scootershare: {},
+    drove_alone: {},
+  });
+});
```

Comment:
Use `expect.any(Type)` for these because empty objects will match literally anything

```suggestion
it('gets the custom mets', async () => {
  initCustomDatasetHelper(getConfig());
  await new Promise((r) => setTimeout(r, 800));
  expect(getCustomMETs()).toMatchObject({
    walk: expect.any(Object),
    bike: expect.any(Object),
    bikeshare: expect.any(Object),
    'e-bike': expect.any(Object),
    scootershare: expect.any(Object),
    drove_alone: expect.any(Object),
  });
});

it('gets the custom footprint', async () => {
  initCustomDatasetHelper(getConfig());
  await new Promise((r) => setTimeout(r, 800));
  expect(getCustomFootprint()).toMatchObject({
    walk: expect.any(Number),
    bike: expect.any(Number),
    bikeshare: expect.any(Number),
    'e-bike': expect.any(Number),
    scootershare: expect.any(Number),
    drove_alone: expect.any(Number),
  });
});
```

Reply:
This uses setTimeout too – can you change it to properly `await getConfig()`?

Reply:
I made the change to get rid of timeout first, so I couldn't commit from the suggestion, added the expect types here: [3dee26f](https://github.com/e-mission/e-mission-phone/pull/1098/commits/3dee26f69f6a54f5265d938ac0897e713bb385cc)